### ISSUE TITLE
RedisClient does not use database from configuration when used with sentinel

### ISSUE THREAD

**Issue by sap-nocops** [quarkusio/quarkus#18742]
### Describe the bug

I use `io.quarkus.redis.client.RedisClient` to read and write from/to redis and setting the host and database via `quarkus.redis.hosts`.
when I use the standalone redis everything works as expected.
In production I also specify `quarkus.redis.client-type` as `sentinel` and read and write works but on databse `0` which is not the one I specified in `quarkus.redis.hosts`

### Expected behavior

RedisClient uses the database specified in `quarkus.redis.hosts` when in sentinel mode

### Actual behavior

RedisClient uses database `0` independently from what is specified in `quarkus.redis.hosts` when in sentinel mode

### How to Reproduce?

set appication property as 
```
quarkus.redis.hosts=redis://host1:26379/9,redis://host2:26379/9,redis://host3:26379/9
quarkus.redis.client-type=sentinel
```
in code do
```
@Inject
RedisClient client;
...

client.set(List.of("key", "value"));
```

you'll find the key in database `0` instead of `9`

### Output of `uname -a` or `ver`

docker image quay.io/quarkus/quarkus-distroless-image:1.0

### Output of `java -version`

11

### GraalVM version (if different from Java)

21.1-java11

### Quarkus version or git rev

2.0.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle 7.1.1

### Additional information

redis and quarkus are running on kubernetes

**Comment by geoand** [comment]
Also cc @pmlopes as this might be a Vert.x Redis Client issue

**Comment by gsmet** [comment]
Yeah I wouldn't be surprised if the issue was in Vert.x given what I've seen in our code.

**Comment by pmlopes** [comment]
While the `SELECT` command is explicit about "single" and "cluster": https://redis.io/commands/select modes, nothing is stated about "sentinel". For this reason, the vertx client did not consider the db in the connection. Users can manually issue a "SELECT 9" to overcome the current limitation.

Meanwhile, a new issue should be added upstream to track the support for allowing db selects in sentinel mode. It is important to note that the sentinel itself cannot really change the database, as it has no meaning. The watcher is watching the availability of the server, not really the database.

**Comment by machi1990** [comment]
@sap-nocops can you please see https://github.com/vert-x3/vertx-redis-client/pull/305#issuecomment-888898860, thanks ?

**Comment by Ladicek** [comment]
This should have been fixed in upstream a long time ago. Please reopen if this is still an issue.
